packages/client/hmi-client/src/workflow/ops/funman/tera-funman.vue

```diff
@@ -0,0 +1,436 @@
+});
+
+const runMakeQuery = async () => {
+	if (!model.value) {
+		toast.error('', 'No Model provided for request');
+		return;
+	}
+
+	const request: FunmanPostQueriesRequest = {
+		model: model.value,
+		request: {
+			constraints: requestConstraints.value,
+			parameters: requestParameters.value,
+			structure_parameters: [
+				{
+					name: 'schedules',
+					schedules: [
+						{
+							timepoints: requestStepList.value
+						}
+					]
+				}
+			],
+			config: {
+				use_compartmental_constraints: true,
+				normalization_constant: 1,
+				tolerance: tolerance.value
+			}
+		}
+	};
+
+	const response = await makeQueries(request); // Just commented out so i do not break funman
+	getStatus(response.id);
+};
+
+// TODO: Poller? https://github.com/DARPA-ASKEM/terarium/issues/2196
+const getStatus = async (runId) => {
+	showSpinner.value = true;
+	const response = await getQueries(runId);
+	if (response?.error === true) {
+		showSpinner.value = false;
+		toast.error('', 'An error occured Funman');
+		console.log(response);
+	} else if (response?.done === true) {
+		showSpinner.value = false;
+		updateOutputPorts(runId);
+	} else {
+		setTimeout(async () => {
+			getStatus(runId);
+		}, 2000);
+	}
+};
+
+const updateOutputPorts = async (runId) => {
+	const portLabel = props.node.inputs[0].label;
+	workflowEventBus.emit('append-output-port', {
+		node: props.node,
+		port: {
+			id: uuidv4(),
+			label: `${portLabel} Result`,
+			type: FunmanOperation.outputs[0].type,
+			value: runId
+		}
+	});
+};
+
+const addConstraintForm = () => {
+	const state = _.cloneDeep(props.node.state);
+	const newGroup: ConstraintGroup = {
+		borderColour: '#00c387',
+		name: '',
+		timepoints: { lb: 0, ub: 100 },
+		variables: []
+	};
+	if (!state.constraintGroups) {
+		state.constraintGroups = [];
+	}
+	state.constraintGroups.push(newGroup);
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+const deleteConstraintGroupForm = (data) => {
+	const state = _.cloneDeep(props.node.state);
+	if (!state.constraintGroups) {
+		return;
+	}
+	state.constraintGroups.splice(data.index, 1);
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+const updateConstraintGroupForm = (data) => {
+	const state = _.cloneDeep(props.node.state);
+	if (!state.constraintGroups) {
+		return;
+	}
+	state.constraintGroups[data.index] = data.updatedConfig;
+
+	workflowEventBus.emitNodeStateChange({
+		workflowId: props.node.workflowId,
+		nodeId: props.node.id,
+		state
+	});
+};
+
+// Used to set requestStepList.
+// Grab startTime, endTime, numberOfSteps and create list.
+function getStepList() {
+	const aList = [startTime.value];
+	const stepSize = (endTime.value - startTime.value) / numberOfSteps.value;
+	for (let i = 1; i < numberOfSteps.value; i++) {
+		aList[i] = i * stepSize;
+	}
+	aList.push(endTime.value);
+	return aList;
+}
```

Comment:
This algorithm will  allow for steps that are not integers, is that okay with Funman or will we need to round/floor/ceil the numbers?

Reply:
Good catch! I will look into this

Reply:
timestep will be integer only

startTime, numberOfSteps, endTime are restricted to integers by the UI however the type is `number`
adding floor to stepSize